MIGRATION.md

```diff
@@ -302,6 +304,14 @@
   - [Packages renaming](#packages-renaming)
   - [Deprecated embedded addons](#deprecated-embedded-addons)
 
+## From version 7.3.0 to 7.4.0
+
+#### CommonJS with Vite is deprecated
+
+Using CommonJS in the `main` configuration with `main.cjs` or `main.cts` is deprecated, and will be removed in Storybook 8.0. This is a necessary change because Vite will remove support for CommonJS in the upcoming v5 release.
+
+Rename your `main` configuration file to `main.js` or `main.ts` and refactor any CommonJS syntax - like `require()` or `module.exports` - to ESM.
```

Comment:
What you suggested will only work if the user already has `"type": "module"` in their `package.json"`
```suggestion
The ideal long-term solution is to add `"type": "module"` to your package.json and convert your project to ESM. To convert to ESM you will need to replace any CommonJS syntax in `.js` files like `require()`, `module.exports`, or `__dirname` or rename those files to use a `.cjs` or `.cts` extension . However, in the short-term you can convert just your `main` configuration file to ESM syntax and rename it to `main.mjs` or `main.mts`.
```

Reply:
I'm reluctant to tell the user to do all of this, it's intimidating or difficult for many to do, and I don't think Storybook should be the one to tell them to do it.  It's much less disruptive for them to simply rename the file to `.mts/.mjs`.

Reply:
Fair. We could also tweak the wording to be a bit softer. E.g. suggesting the mjs route first and softening up some of the language

Reply:
I updated the suggested wording to be a bit softer. I think it all needs to stay except for maybe the last sentence, which is only a possible suggestion

Reply:
Seems fine to me.  Though it's also possible that their project is already ESM, and they just had a legacy `.cjs` extension for Storybook because it didn't support ESM config until v7.  So perhaps we could say something like,

```
If you haven't already, you may also consider adding `"type": "module"` to your package.json and converting your project to ESM.
```

Reply:
> If you haven't already, you may also consider adding `"type": "module"` to your package.json and converting your project to ESM.

This would be a lot harder for users than just converting the `main` config file

Reply:
Yes, I was just talking about that last sentence in your proposal.  I think the rest should stay as you have it.

Reply:
Ah, okay! That makes sense. I've update the proposal with that